crates/client/src/client.rs

```diff
@@ -51,10 +42,16 @@ impl MojaveClient {
     async fn send_request_race(
         &self,
         request: RpcRequest,
+        target_addresses: &[String],
     ) -> Result<RpcResponse, MojaveClientError> {
-        let requests: Vec<Pin<Box<Fuse<_>>>> = self
-            .inner
-            .urls
+        let urls = target_addresses
```

Comment:
Check if urls is empty and return an error in that case.

could we change that cause there we are parsing the url on each call which is not really good, same for every place where url::parse is called this shall be called only once